src/linear/updater_gpu_coordinate.cu

```diff
@@ -58,10 +58,15 @@ class GPUCoordinateUpdater : public LinearUpdater {  // NOLINT
 
     num_row_ = static_cast<size_t>(p_fmat->Info().num_row_);
 
-    CHECK(p_fmat->SingleColBlock());
-    SparsePage const& batch = *(p_fmat->GetBatches<CSCPage>().begin());
+    auto batch_iter = p_fmat->GetBatches<CSCPage>().begin();
```

Comment:
I'm not sure why you want to remove this method and then created two different approaches to obtain the check.  One in gbm with dynamic cast another one here.

Reply:
I can use the dynamic cast approach here for consistency.

Reply:
It seems that single method is more simple, we don't need to include extra header here, it breaks the abstraction.

Reply:
I reverted the change. Note that the abstraction is broken just by having a method in a base class that differentiates behaviour based on the subclass.